File: app/src/main/java/com/wkz/progressbanner/sample/bean/SimpleBannerModel.java
package com.wkz.progressbanner.sample.bean;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.wkz.bannerlayout.listener.BannerModelCallBack;

public class SimpleBannerModel implements BannerModelCallBack {
    @Nullable
    private Object image;
    @Nullable
    private String title;

    @Nullable
    public final Object getImage() {
        return this.image;
    }

    public final void setImage(@Nullable Object var1) {
        this.image = var1;
    }

    @Nullable
    public final String getTitle() {
        return this.title;
    }

    public final void setTitle(@Nullable String var1) {
        this.title = var1;
    }

    @NonNull
    public String getBannerUrl() {
        return String.valueOf(this.image);
    }

    @NonNull
    public String getBannerTitle() {
        return String.valueOf(this.title);
    }

    public SimpleBannerModel(@NonNull Object image) {
        super();
        this.image = image;
    }

    public SimpleBannerModel(@NonNull Object image, @NonNull String title) {
        super();
        this.image = image;
        this.title = title;
    }

    public SimpleBannerModel() {
    }
}

File: app/src/main/java/com/fphoenixcorneae/progressbanner/sample/SimpleBannerModel.kt
package com.fphoenixcorneae.progressbanner.sample

import com.fphoenixcorneae.bannerlayout.listener.BannerModelCallBack

class SimpleBannerModel : BannerModelCallBack<String?> {

    var image: Any? = null

    override var bannerTitle: String? = null

    override val bannerUrl: String
        get() = image.toString()

    constructor(image: Any) : super() {
        this.image = image
    }

    constructor(image: Any, title: String) : super() {
        this.image = image
        bannerTitle = title
    }

    constructor() {}
}